dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -251,51 +258,77 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
                 registeredTaskFromBuilder = builder.Register();
             }
 
-            PushNotificationRegistrationToken token = { cookie, registeredTaskFromBuilder };
             scopeExitToCleanRegistrations.release();
 
-            PushNotificationTelemetry::ActivatorRegisteredByApi(S_OK, details.Options());
+            auto lock = s_activatorInfoLock.lock_exclusive();
+            s_cookie.reset(cookie);
```

Comment:
What's the reason of doing this in two steps instead of this?

`CoRegisterClassObject(..., &s_cookie);`

Also, probably it's not `s_cookie` but `s_comActivatorRegistration` or similar;

Reply:
In the previous implementation, we didn't want to return the s_cookie unless we made it to the end of the function. We can just call s_cookie.reset() in the scope_exit.